src/settings.ts

```diff
@@ -63,5 +67,85 @@ export class LinkRangeSettingTab extends PluginSettingTab {
 					await this.plugin.saveSettings();
 					postProcessorUpdate(this.app)
 				}));
+	
+		new Setting(this.containerEl)
+			.setName("Add New")
+			.setDesc("Add new pattern to match files in a directory. The first value will change the visual for the heading in a link. The second value will change the visual for separator. The third specifies the folder in which the files must be to match.")
+			.addButton((button: ButtonComponent) => {
+				button
+					.setTooltip("Add new pattern to match files in a directory.")
+					.setButtonText("+")
+					.setCta()
+					.onClick(() => {
+						this.plugin.settings.patterns.push({
+							headingVisual: '',
+							headingSeparatorVisual: '',
+							path: ''
+						});
+						this.plugin.saveSettings();
+						this.display();
+					});
+			});
+		
+			this.plugin.settings.patterns.forEach(
+				(pattern, index) => {
+					const s = new Setting(this.containerEl)
+					.addText(text => text
+						.setPlaceholder('Enter a heading override')
+						.setValue(pattern.headingVisual)
+						.onChange(async (value) => {
+							pattern.headingVisual = value;
+							await this.plugin.saveSettings();
+							postProcessorUpdate(this.app)
+						}))
+					.addText(text => text
+						.setPlaceholder('Enter a separator override')
+						.setValue(pattern.headingSeparatorVisual)
+						.onChange(async (value) => {
+							pattern.headingSeparatorVisual = value;
+							await this.plugin.saveSettings();
+							postProcessorUpdate(this.app)
+						}))
+					.addText(text => text
+						.setPlaceholder('Enter a path')
+						.setValue(pattern.path)
+						.onChange(async (value) => {
+							pattern.path = value;
+							await this.plugin.saveSettings();
+							postProcessorUpdate(this.app)
+						}));
+
+					if (index === 0) {
+						s.addExtraButton((cb) => {
+							cb.setIcon("lock")
+								.setTooltip("This pattern is the default and cannot be completed");					
```

Comment:
Should this say "This pattern is the default and cannot be **deleted**?

Reply:
Yep